### PR TITLE
Fix newly added entry not synced to db

### LINKED ISSUE #6944: JabRef not syncing changes to shared databases (Mysql/MariaDB and Postgresql)
JabRef 5.1--ArchLinux--1, Linux 5.8.10-arch1-1 amd64, Java 14.0.2

I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists:
Dev Version tested:
JabRef 5.2--2020-09-24--75b6770, Linux 5.8.10-arch1-1 amd64, Java 15
(The same problem exists with the latest MAC Version)

Databases tested:
PostgreSQL 12 (local)
PostgreSQL instance from elephantsql
MariaDB 10.1.40 (remote)

After successful connection to an empty Database the tables are created as expected but any entries made in JabRef are not being synced. The input of a new entry instead immedeatly results in the errors attached bellow. 
Interesstingly when importing an existing library to the shared database the entries are created as expected.  

Steps to reproduce the behaviour:
1. Set up a database and connect to it in JabRef
2. Add some Entry in JabRef

ps Thank you for all the great work!
pps Sorry if I should have missed something obvious and stupid.

<details>
  <summary>Log File</summary>
Connection to PostgreSQL server established.
Enabling live reloading of /usr/share/java/jabref/resources/main/org/jabref/gui/Base.css
Enabling live reloading of /usr/share/java/jabref/resources/main/org/jabref/gui/Dark.css
SQL Error: 
org.postgresql.util.PSQLException: No value specified for parameter 1.
	at org.postgresql.jdbc@42.2.16/org.postgresql.core.v3.SimpleParameterList.checkAllParametersSet(SimpleParameterList.java:275)
	at org.postgresql.jdbc@42.2.16/org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:307)
	at org.postgresql.jdbc@42.2.16/org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:473)
	at org.postgresql.jdbc@42.2.16/org.postgresql.jdbc.PgStatement.execute(PgStatement.java:393)
	at org.postgresql.jdbc@42.2.16/org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:164)
	at org.postgresql.jdbc@42.2.16/org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:130)
	at org.jabref@5.1/org.jabref.logic.shared.DBMSProcessor.insertIntoFieldTable(DBMSProcessor.java:274)
	at org.jabref@5.1/org.jabref.logic.shared.DBMSProcessor.insertEntries(DBMSProcessor.java:155)
	at org.jabref@5.1/org.jabref.logic.shared.DBMSSynchronizer.listen(DBMSSynchronizer.java:85)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at com.google.common@29.0-jre/com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:87)
	at com.google.common@29.0-jre/com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:144)
	at com.google.common@29.0-jre/com.google.common.eventbus.Subscriber$1.run(Subscriber.java:72)
	at com.google.common@29.0-jre/com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at com.google.common@29.0-jre/com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:67)
	at com.google.common@29.0-jre/com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:108)
	at com.google.common@29.0-jre/com.google.common.eventbus.EventBus.post(EventBus.java:212)
	at org.jabref@5.1/org.jabref.model.database.event.CoarseChangeFilter.listen(CoarseChangeFilter.java:29)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at com.google.common@29.0-jre/com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:87)
	at com.google.common@29.0-jre/com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:144)
	at com.google.common@29.0-jre/com.google.common.eventbus.Subscriber$1.run(Subscriber.java:72)
	at com.google.common@29.0-jre/com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at com.google.common@29.0-jre/com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:67)
	at com.google.common@29.0-jre/com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:108)
	at com.google.common@29.0-jre/com.google.common.eventbus.EventBus.post(EventBus.java:212)
	at org.jabref@5.1/org.jabref.model.database.BibDatabase.insertEntries(BibDatabase.java:206)
	at org.jabref@5.1/org.jabref.model.database.BibDatabase.insertEntries(BibDatabase.java:195)
	at org.jabref@5.1/org.jabref.gui.BasePanel.insertEntries(BasePanel.java:254)
	at org.jabref@5.1/org.jabref.gui.BasePanel.insertEntry(BasePanel.java:240)
	at org.jabref@5.1/org.jabref.gui.importer.NewEntryAction.execute(NewEntryAction.java:57)
	at org.jabref@5.1/org.jabref.gui.actions.JabRefAction.lambda$new$2(JabRefAction.java:41)
	at org.controlsfx.controls/org.controlsfx.control.action.Action.handle(Action.java:419)
	at org.controlsfx.controls/org.controlsfx.control.action.Action.handle(Action.java:64)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics/javafx.scene.Node.fireEvent(Node.java:8885)
	at javafx.controls/javafx.scene.control.Button.fire(Button.java:203)
	at javafx.controls/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:206)
	at javafx.controls/com.sun.javafx.scene.control.inputmap.InputMap.handle(InputMap.java:274)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.process(Scene.java:3890)
	at javafx.graphics/javafx.scene.Scene.processMouseEvent(Scene.java:1885)
	at javafx.graphics/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2618)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:409)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:299)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:447)
	at javafx.graphics/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:412)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:446)
	at javafx.graphics/com.sun.glass.ui.View.handleMouseEvent(View.java:556)
	at javafx.graphics/com.sun.glass.ui.View.notifyMouse(View.java:942)
	at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
	at java.base/java.lang.Thread.run(Thread.java:832)
</details>

### LINKED ISSUE #7176: Shared Databases: creating new entries does not work, importing existing libraries does
Disclaimer: I hope I do not missed important information within the tutorials...

Setup:
- OS: Windows10 64-Bit
- JabRef version tested: 5.1 and 5.2 (09.12.2020) against PostgreSQL
- PostgreSQL: 12 and 13
- Java 1.8.0_261-b12

Steps done:
- starting the postgreSQL service -> creating database ->creating user for database (all done by pgadmin4, localhost)
- starting JabRef -> open shared database -> save to local backup -> changing standard file directory -> start creating entries with"plus"-button (articles)

Expected behaviour:
Two new entries are created and I filled in the minimal needed information for articles. Both are shown correctly within JabRefs UI. I saved the entries and closed JabRef. There were no error messages shown. Expected behaviour: after reopneing the shared database my entries are loaded and presented and the backup-file is written accordingly.

What happened:
After restarting JabRef automatically connected to the PostgreSQL server. However: there are no entries inside the shared database as well as the backup-file.

What I tried:
I tried importing a locally stored database into the shared one. This worked as expected. After closing and opening the shared-database all imported entries are shown and can be edited. However still no new single entries can be created or edited.

I checked the PostgreSQL databases content and had the following findings:
At the first connection between JabRef and PostgreSQL three tables are created

- public.ENTRY (shared_id | type | version)
- public.FIELD (entry_shared_id | name | value
- public.METADATA (key  value)

METADATA is written and restored correctly (database type and fileDirectory).
ENTRY is written correctly, if the "plus"-button is used shared_id is increased by one, type is updated and version is set to one. Works with import feature to.
FIELD is written correctly if the import to database feature is used: entry_shared_id equals  ENTRY.shared_id and the entries are created. However, if "create new entry" is used nothing is written to FIELD, but only to ENTRY. The import feature respects the shared_ids and numerates according to ENTRY.

In consequence this leads to empty databases and backup-files if the import local database feature is not used.

Maybe this is connected to #7159 or the Java version? It is still unclear to me, why it works using the import feature but does not when creating single entries.

I am unsure where to get the logfile if no error is given.

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes #7176 
Should also fix #6944 

Newly added entries have empty fields; don't update the field table to prevent SQL Exception
Fix shared entry not found by id
use left outer join for this

- [x] Change in CHANGELOG.md described (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
I don't know why the documentation test fails

File: src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java [hunk 1]
```diff
@@ -95,6 +95,37 @@ void testInsertEntry() throws SQLException {
...
+            assertFalse(entryResultSet.next());
+
+            try (ResultSet fieldResultSet = selectFrom("FIELD", dbmsConnection, dbmsProcessor)) {
+                while (fieldResultSet.next()) {
+                    actualFieldMap.put(fieldResultSet.getString("NAME"), fieldResultSet.getString("VALUE"));
+                }
+            }
+        }
+
+        Map<String, String> expectedFieldMap = expectedEntry.getFieldMap().entrySet().stream().collect(Collectors.toMap((entry) -> entry.getKey().getName(), Map.Entry::getValue));
```
Comment (tobiasdiez) [inline src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java:124]: Isn't this always an empty list (as expected entry is empty)?

File: src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java [hunk 2]
```diff
@@ -95,6 +95,37 @@ void testInsertEntry() throws SQLException {
...
 
+    @Test
+    void testInsertEntryWithEmptyFields() throws SQLException {
+        BibEntry expectedEntry = new BibEntry(StandardEntryType.Article);
+
+        dbmsProcessor.insertEntry(expectedEntry);
+
+        BibEntry emptyEntry = getBibEntryExample();
+        emptyEntry.getSharedBibEntryData().setSharedID(1);
+        dbmsProcessor.insertEntry(emptyEntry); // does not insert, due to same sharedID.
```
Comment (tobiasdiez) [inline src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java:106]: why is this then necessary?

**Review by tobiasdiez — CHANGES_REQUESTED** [review]

File: src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java (hunk 1 above)
Comment (Siedlerchr) [reply src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java:124]: Hm, seems like I need to adapt the queries a bit. Didn't think about this,
I should check for the shared entry id to be present.

Comment (tobiasdiez) [reply src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java:124]: To be honest, I've no idea what the test is doing. You add an empty entry, then some other completely unrelated entry, and then compare against the empty entry again.

File: src/main/java/org/jabref/logic/shared/DBMSProcessor.java [hunk 3]
```diff
@@ -493,7 +499,7 @@ public void removeEntries(List<BibEntry> bibEntries) {
              .append("F.").append(escape("VALUE"))
              .append(" FROM ")
              .append(escape("ENTRY"))
-             .append(" inner join ")
+             .append(" left outer join ")
```
Comment (koppor) [inline src/main/java/org/jabref/logic/shared/DBMSProcessor.java:502]:
```suggestion
             // Handle special case if entry does not have any fields (yet)
             .append(" left outer join ")
```

**Pull request merged**